examples/react-datastore/src/App.tsx

```diff
@@ -5,30 +5,43 @@ import 'antd/dist/antd.css';
 import { useFindTodos } from './datastore/hooks';
 import { TodoList, AddTodo, Loading, Error, Header } from './components';
 import { datastore } from './datastore/config';
+import { NetworkStatusEvent } from 'offix-datastore/types/replication/network/NetworkStatus';
 
 function App() {
 
   const [replicating, setReplicating] = useState<boolean>(true);
   const [addView, setAddView] = useState<boolean>(false);
   const  { loading, error, data, subscribeToUpdates } = useFindTodos();
 
+  useEffect(() => {
+    datastore.getNetworkIndicator()?.subscribe({
+      next: (event: NetworkStatusEvent) => {
+        if (event.isOnline) {
+          datastore.startReplication();
+          setReplicating(true);
+        } else {
+          datastore.stopReplication();
+          setReplicating(false);
+        }
+      }
+    })
+  })
+
   useEffect(() => {
     // We can start replication on a per model basis
     // or for the entire store with:
     // datastore.startReplication
     // the `startReplication` method accepts an
     // optional filter
-    if (replicating) {
-      datastore.startReplication()
-    }
-  }, [replicating]);
+    datastore.startReplication()
+  });
 
-  const toggleReplication = () => {
-    if (replicating) {
-      datastore.stopReplication()
-    } 
-    setReplicating(!replicating)
-  }
+  // const toggleReplication = () => {
```

Comment:
Why this is disabled

Reply:
Oh should have deleted. I removed it now because I have the change in network status stopping and starting the replication automatically

Reply:
That should happen anyway in the core right?

Reply:
Yeah, this is more for the OVP stuff, because turning switching to offline mode in chrome doesn't really stop the replication or subscriptions from coming in.